docs/howto/script_testing.md

```diff
@@ -0,0 +1,209 @@
+# HOWTO: Writing script tests for a package
+
+Script testing is an advanced topic that assumes knowledge of [pipeline](./pipeline_testing.md)
+and [system](./system_testing.md) testing.
+
+Testing packages with script testing is only intended for testing cases that
+cannot be adequately covered by the pipeline and system testing tools such as
+testing failure paths and package upgrades. It can also be used for debugging
+integrations stack issues.
+
+## Introduction
+
+The script testing system is build on the Go testscript package with extensions
+provided to allow scripting of stack and integration operations such as
+bringing up a stack, installing packages and running agents. For example, using
+these commands it is possible to express a system test as described in the
+system testing [Conceptual Process](./system_testing.md#conceptual-process) section.
+
+
+## Expressing tests
+
+Tests are written as [txtar format](https://pkg.go.dev/golang.org/x/tools/txtar#hdr-Txtar_format)
+files in a data stream's \_dev/test/scripts directory. The logic for the test is
+written in the txtar file's initial comment section and any additional resource
+files are included in the txtar file's files sections.
+
+The standard commands and behaviors for testscript scripts are documented in
+the [testscript package documentation](https://pkg.go.dev/github.com/rogpeppe/go-internal/testscript).
```

Comment:
Just for the record. Could you comment, maybe in the description of the PR, why you chose `testscript` instead of some other embedded language?

Reply:
It's the only well maintained and extensively used script testing package available with the features that we need. It's not a language, so I'm not sure what competing systems you are referring to.